recap/examples/log.rs

```diff
@@ -21,8 +21,10 @@ fn main() -> Result<(), Box<dyn Error>> {
   2 false world"#;
 
     for line in logs.lines() {
-        let entry: LogEntry = line.parse()?;
-        println!("{:#?}", entry);
+        if LogEntry::is_match(line) {
+            let entry: LogEntry = line.parse()?;
+            println!("{:#?}", entry);
+        }
```

Comment:
Do you have a concrete case where this improves performance? I would definitely like to see it, because it's likely a bug in the regex library's heuristics. There are perhaps some cases where this is true, e.g., searching a big haystack where a match is known very early in the haystack.

(Also, if you do wind up needing this, it would be good if there was a way to do it that didn't involve compiling the regex twice.)